src/screens/ChannelListScreenV2/index.tsx

```diff
@@ -26,6 +31,82 @@ const ChannelListScreenV2 = () => {
     navigation.navigate('ContactScreen');
   };
 
+  const requestPermission = async () => {
+    const authStatus = await messaging().requestPermission();
+    const enabled =
+      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+      authStatus === messaging.AuthorizationStatus.PROVISIONAL;
+
+    if (enabled) {
+      const fcmToken = await messaging().getToken();
+      const payload = {
+        fcm_token: fcmToken
+      };
+      fcmTokenService(payload);
+    }
+  };
+
+  React.useEffect(() => {
+    const checkNotif = async () => {
+      if (!navigation.isFocused()) return;
+      PushNotification.checkPermissions(
+        (
+          data: PushNotificationPermissions & {
+            authorizationStatus?: number;
+          }
+        ) => {
+          const lastPromptTime = StorageUtils.lastPromptNotification.get();
+
+          // For testing purpose,  (20000 milliseconds)
+          const promptInterval = 48 * 3600000;
+          const currentTime = new Date().getTime();
+          if (lastPromptTime && currentTime - parseFloat(lastPromptTime) < promptInterval) {
+            return; // Don't show the prompt if the interval has not passed
+          }
+
+          const needsPermission =
+            data.alert === false || data.badge === false || data.sound === false;
+
+          const showAlert = (onPressAction: () => void) =>
+            Alert.alert(
+              "Don't Miss New Messages",
+              `Allow notifications to know 
+when friends send you messages.`,
+              [
+                {text: 'Not now', onPress: () => console.log('Cancel Pressed')},
+                {text: 'Allow', onPress: onPressAction, isPreferred: true}
+              ]
+            );
+
+          switch (data.authorizationStatus) {
+            case 1:
```

Comment:
can we change this to enum or constant?
Like Status.ACCEPTED etc?

Reply:
for sure we can use that. will change that to enum.